packages/flutter/test_private/README.md

```diff
@@ -0,0 +1,51 @@
+# Private Test Runner
+
+These are tests of private interfaces that can't easily happen in the regular
+flutter tests due to problems with test and implementation interdependence.
+
+This gets around the problem of parts existing in more than one library by
+making a copy of the code under test.
+
+The test script `bin/test_private.dart` tests private interfaces by copying the
+code under test into a temporary workspace.  The test is then free to make the
+copied flutter source into a "part" of its own library by declaring a library
+and using the `part` directive with a relative path to include the parts. This
+way the test and the private interface are part of the same library, and the
+private interface can be accessed by the test.
+
+The tests are run like so:
+
+```shell
+pub run --enable-experiment=non-nullable test_private.dart
+```
+
+One limitation is that the copied code needs to be separable enough to be
+copied, so it needs to be in its own separate files, or extracted out of a
+larger file by the script (not a method currently employed).
```

Comment:
let's not encourage that second path if we don't need it... requiring that this approach only be used by entirely separable code seems like a good place to start.

Reply:
Good point.  Removed.